File: DoitMission_12/Doit12_Java/app/src/main/java/com/example/doit12_java/TextReceiver.java
package com.example.doit12_java;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.util.Log;

public class TextReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        String message = intent.getStringExtra(MainActivity.message);
        Intent broadIntent = new Intent(context,MainActivity.class);
        broadIntent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_SINGLE_TOP);
        broadIntent.putExtra(MainActivity.message,message);
        context.startActivity(broadIntent);
        Intent stopService1 = new Intent(context,TextService.class);
        context.stopService(stopService1);
    }
}

File: DoitMission_12/Doit12_Kotlin/app/src/main/java/com/example/doit12_kotlin/TextReceiver.kt
package com.example.doit12_kotlin

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

class TextReceiver : BroadcastReceiver() {

    override fun onReceive(context: Context, intent: Intent) {

        val textMessage = intent.getStringExtra(MainActivity.message)
        val mainIntent = Intent(context,MainActivity::class.java)
        mainIntent.flags = Intent.FLAG_ACTIVITY_NEW_TASK or Intent.FLAG_ACTIVITY_SINGLE_TOP or Intent.FLAG_ACTIVITY_CLEAR_TOP
        mainIntent.putExtra(MainActivity.message,textMessage)
        context.startActivity(mainIntent)
        val spService = Intent(context,TextService::class.java)
        context.stopService(spService)
    }
}
